DragonSlayer.cpp

```diff
@@ -1,28 +1,54 @@
 #include "DragonSlayer.h"
 #include "Dragon.h"
+#include "AttackItem.h"
+#include "Utility.h"
 
 //DragonSlayer::DragonSlayer
 
+DragonSlayer::DragonSlayer(std::string name_, int hp_, int armor_) : Character(hp_, armor_, 4), name(name_) 
+{
+    attackItem = std::unique_ptr<Item>(new AttackItem());
+    helpfulItems =  makeHelpfulItems(4);
+    defensiveItems = makeDefensiveItems(2);
+}
+
 //DragonSlayer::getName
+const std::string& DragonSlayer::getName()
+{
+    return name;
+}
 
 void DragonSlayer::attack(Character& other)
 {
     std::cout << name << " is attacking " << other.getName() << " !!" << std::endl;
     if( auto* dragon = dynamic_cast<Dragon*>(&other) )
     {
-        assert(false);
+        // assert(false);
         //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
         //so they should USE their attack item before attacking the dragon... 
         //note that items are single-use only, so you need to reset it after use.  
         //look in the Character class for how the other item types are reset after use.
+        attackItem.get()->use(this);
+        attackItem.reset();
```

Comment:
make sure your attackItem is valid before you use it.
you don't need to `.get()` to `->use()` it.  unique_ptr<> is a wrapper around pointers and when you learn about overloaded operators, you'll understand how the `->` and `*` work in the unique_ptr class and remove the need to use `.get()`
- [ ] revise